python/Ganga/new_tests/TestProxy.py

```diff
@@ -79,7 +85,7 @@ def test_dict_attributes(self):
         self.assertFalse(hasattr(proxied, '_impl'))
 
         # A proxy instance should only have _impl in its dict
-        self.assertEqual(self.p.__dict__.keys(), ['_impl'])
+        self.assertEqual(self.p.__dict__.keys(), ['_impl', '_proxyObject'])
```

Comment:
Before your changes, line 73 was:
```python
self.assertFalse(hasattr(self.p, '_proxyObject'))
```
because `self.p` is a proxy object wrapping a `TestGangaObject` instance. It should not have a `_proxyObject` since it is itself a proxy.

If you are seeing a proxy object with a `_proxyObject` attribute then the code needs to be fixed, not the test.

If
```python
self.assertFalse(hasattr(self.p, '_proxyObject'))
```
then (it doesn't have `_proxyObject`) and so
```python
self.assertEqual(self.p.__dict__.keys(), ['_impl'])
```
should be fine?